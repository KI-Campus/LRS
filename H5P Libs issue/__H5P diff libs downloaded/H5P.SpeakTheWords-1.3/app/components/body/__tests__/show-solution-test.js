jest.unmock('../show-solution');

import React from 'react';
import ShowSolution from '../show-solution';
import renderer from 'react-test-renderer';

describe('Show Solution', () => {
  const eventStoreMock = {on: jest.fn(), trigger: jest.fn()};
  const params = {
    acceptedAnswers: ['Dart', 'Vader'],
    l10n: {
      correctAnswersText: 'Correct',
      userAnswersText: 'User answers'
    }
  };

  const test = (action) => {
    const component = renderer.create(
      <ShowSolution
        eventStore={eventStoreMock}
        {...params}
      />
    );

    let tree = component.toJSON();
    action(tree, component);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  };

  it('should be hidden initially', () => {
    test(() => {});
  });

  it('should be shown on show solution event', () => {
    test((tree, component) => {
      component.getInstance().setState({showSolution: true});
    })
  });

  it('should contain user answers when specified', () => {
    test((tree, component) => {
      component.getInstance().setState({
        showSolution: true,
        userAnswers: ['Luke', 'SkyWalker']
      });
    });
  });
});
