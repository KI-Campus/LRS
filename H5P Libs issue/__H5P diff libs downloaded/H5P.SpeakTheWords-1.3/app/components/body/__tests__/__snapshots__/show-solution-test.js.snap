// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Solution should be hidden initially 1`] = `
<div
  className="h5p-speak-the-words-solution-area hidden"
>
  <div
    className="h5p-speak-the-words-correct-answer-text"
  >
    <div>
      Correct
    </div>
    <div
      className="h5p-speak-the-words-correct-answer"
    >
      Dart
    </div>
    <div
      className="h5p-speak-the-words-correct-answer"
    >
      Vader
    </div>
  </div>
</div>
`;

exports[`Show Solution should be shown on show solution event 1`] = `
<div
  className="h5p-speak-the-words-solution-area"
>
  <div
    className="h5p-speak-the-words-correct-answer-text"
  >
    <div>
      Correct
    </div>
    <div
      className="h5p-speak-the-words-correct-answer"
    >
      Dart
    </div>
    <div
      className="h5p-speak-the-words-correct-answer"
    >
      Vader
    </div>
  </div>
</div>
`;

exports[`Show Solution should contain user answers when specified 1`] = `
<div
  className="h5p-speak-the-words-solution-area"
>
  <div
    className="h5p-speak-the-words-user-answer-text"
  >
    <div>
      User answers
    </div>
    <div
      className="h5p-speak-the-words-interpreted-answer"
    >
      Luke
    </div>
    <div
      className="h5p-speak-the-words-interpreted-answer"
    >
      SkyWalker
    </div>
  </div>
  <div
    className="h5p-speak-the-words-correct-answer-text"
  >
    <div>
      Correct
    </div>
    <div
      className="h5p-speak-the-words-correct-answer"
    >
      Dart
    </div>
    <div
      className="h5p-speak-the-words-correct-answer"
    >
      Vader
    </div>
  </div>
</div>
`;
